linker/shared/src/test/scala/org/scalajs/linker/AnalyzerTest.scala

```diff
@@ -297,6 +297,50 @@ class AnalyzerTest {
     }
   }
 
+  @Test
+  def missingInheritedMethod(): AsyncResult = await {
+    /* Test that we do not use an inherited missing method:
+     * Otherwise we'll open ourselves up to non-deterministic behavior that
+     * depends on the call graph traversal order.
+     *
+     * The invocation below is carefully crafted to delay traversal of the call
+     * to B#foo as late as possible. If the lookup of B#foo would re-use the
+     * created missing method in A#foo, we'd get one error less.
+     */
+
+    val fooMethodName = m("foo", Nil, V)
+    val barMethodName = m("bar", Nil, V)
+
+    val classDefs = Seq(
+        classDef("A", superClass = Some(ObjectClass),
+            methods = List(
+                trivialCtor("A"),
+                MethodDef(EMF, barMethodName, NON, Nil, NoType, Some(Block(
+                  Apply(EAF, This()(ClassType("A")), fooMethodName, Nil)(NoType),
+                  Apply(EAF, New("B", NoArgConstructorName, Nil), fooMethodName, Nil)(NoType),
+                )))(EOH, UNV)
+            )),
+        classDef("B", superClass = Some("A"),
+            methods = List(trivialCtor("B"))),
```

Comment:
I think Scalastyle won't be happy with the trailing comma here. :(

Reply:
Ah ah, yes. I've removed this one and another one above (in the Block).